leabra7/projn.py

```diff
@@ -53,6 +53,47 @@ def expand_layer_mask_full(pre_mask: List[bool],
     return torch.ger(torch.ByteTensor(post_mask), torch.ByteTensor(pre_mask))
 
 
+def expand_layer_mask_one_to_one(pre_mask: List[bool],
+                                 post_mask: List[bool]) -> torch.ByteTensor:
+    """
+    Expands layer masks into a weight matrix mask
+    with one-to-one connectivity.
+
+    Args:
+        pre_mask: The mask for the pre layer specifying which pre layer
+            units are included in the projection. Note that this mask will not
+            be tiled, so it has as many elements as pre layer units.
+
+        post_mask: The mask for the post layer specifying which post layer
+            units are included in the projection. Has as many elements as post
+            layer units.
+
+    Returns:
+        A mask for the one-to-one projection weight matrix indicating
+        which elements of the matrix correspond to active connections
+        in the full connectivity pattern.
+
+    """
+    if sum(pre_mask) != sum(post_mask):
+        raise ValueError(
+            """Mismatched one-to-one projection. Pre_mask units: {0}.
+            Post_mask units: {1}.""".format(sum(pre_mask), sum(post_mask)))
+
+    mask = torch.zeros(len(pre_mask), len(post_mask)).byte()
+    i = j = 0
```

Comment:
compact but generally considered bad style (it can bite you if the value is a pointer)